operator/internal/central/extensions/common.go

```diff
@@ -59,6 +59,12 @@ func wrapExtension(runFn func(ctx context.Context, central *platform.Central, cl
 			return err
 		}
 
+		// Propagate any annotation changes back to the unstructured object so that
+		// subsequent phases see them
+		if anns := c.GetAnnotations(); anns != nil {
+			u.SetAnnotations(anns)
```

Comment:
I don't feel great about keeping state here.
1. I'm worried the updater might cause this to be [persisted](https://github.com/stackrox/helm-operator/blob/1b2840226a8b2eb457d3b7f36565d2a4744d5e3a/pkg/reconciler/internal/updater/updater.go#L128) to the cluster.
2. It's not clear to me what the interaction with the defaulting flows that explicitly add annotations on the CR would be 🤔 cc @mclasmeier